sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -352,10 +352,7 @@ object BooleanSimplification extends Rule[LogicalPlan] with PredicateHelper {
         val lhs = splitDisjunctivePredicates(left)
```

Comment:
Could you update the comment about how to handle this case?

https://github.com/apache/spark/pull/31318/files#diff-d43484d56a4d9991066b5c00d12ec2465c75131e055fc02ee7fb6dfd45b5006fR347-R351

Reply:
updated the comment, thanks.